fission/src/mirabuf/MirabufLoader.ts

```diff
@@ -238,8 +253,15 @@ class MirabufCachingService {
                 thumbnailStorageID: thumbnailStorageID ?? defaultStorageID,
             }
             map[key] = info
-            miraType == MiraType.ROBOT ? (backUpRobots[id] = info) : (backUpFields[id] = info)
-            window.localStorage.setItem(miraType == MiraType.ROBOT ? robotsDirName : fieldsDirName, JSON.stringify(map))
+            miraType === MiraType.ROBOT
+                ? (backUpRobots[id] = info)
+                : miraType === MiraType.FIELD
+                  ? (backUpFields[id] = info)
+                  : (backUpPieces[id] = info)
+            window.localStorage.setItem(
+                miraType == MiraType.ROBOT ? robotsDirName : miraType == MiraType.FIELD ? fieldsDirName : piecesDirName,
```

Comment:
Stylistically and from a readability standpoint I think that instead of a double ternary, having a dictionary for MiraType to backup object and directory name (and anything else that's in that same scope) instead of separate variables would make this a lot cleaner